### PR TITLE
feat(presets): Implement Publish/Make Default/Delete

### PR DESCRIPTION
**Pull request by glennkitchellcaci**
Improved UX. Integrated calls to PresetServices to Publish/Make Default/Delete.

### CODE REVIEW

File: src/os/layer/preset/presetmenubutton.js [hunk 1]
```diff
@@ -35,12 +38,16 @@ const EventType = {
...
   SAVE: 'save',
+  TOGGLE_PRESET: 'toggle-preset',
   TOGGLE_DEFAULT_TRUE: 'toggle-default-true',
   TOGGLE_DEFAULT_FALSE: 'toggle-default-false',
   TOGGLE_PUBLISHED_TRUE: 'toggle-published-true',
   TOGGLE_PUBLISHED_FALSE: 'toggle-published-false'
 };
 
+/**
+ * @const {GoogLog.Logger}
```
Comment (jsalankey) [inline src/os/layer/preset/presetmenubutton.js:49]: I think this is mixing up `@type {...}` with `@const` which should be a standalone

File: src/os/layer/preset/presetmenubutton.js [hunk 2]
```diff
@@ -245,9 +282,60 @@ class Controller extends MenuButtonCtrl {
...
+    // Refresh the application
+    ConfirmUI.launchConfirm(/** @type {osx.window.ConfirmOptions} */ ({
+      yesText: 'Delete',
+      yesIcon: 'fa fa-trash-o',
+      yesButtonClass: 'btn-danger',
+      confirm: (() => {
+        // get PresetService(s) that support remove()
+        this.getService(OsLayerPreset.PresetServiceAction.REMOVE).then(
+            (service) => {
+              this['thinking'] = true;
```
Comment (jsalankey) [inline src/os/layer/preset/presetmenubutton.js:301]: Is the `this` context on this function correct? It doesn't look like the context argument was provided to the `.then()` call.

File: src/os/layer/preset/presetmenubutton.js [hunk 3]
```diff
@@ -258,14 +346,17 @@ class Controller extends MenuButtonCtrl {
...
-    }
+    // get PresetService(s) that support uodate()
+    this.getService(OsLayerPreset.PresetServiceAction.UPDATE).then(
+        (service) => {
+          service.update(preset).then(
+              this.saveSuccess.bind(this),
+              this.onServiceFailure.bind(this, 'Could not save Preset'));
+        },
+        (msg) => {
+          GoogLog.error(LOGGER, 'No services found that support save action');
```
Comment (jsalankey) [inline src/os/layer/preset/presetmenubutton.js:357]: This should include the `msg` argument value in its logged message.

File: src/os/layer/preset/presetmenubutton.js (hunk 1 above)
Comment (glennkitchellcaci) [reply src/os/layer/preset/presetmenubutton.js:49]: derp. ty ty

File: src/os/layer/preset/presetmenubutton.js [hunk 4]
```diff
@@ -382,15 +540,18 @@ const directive = () => ({
   controllerAs: 'ctrl',
   template: `
 <div class="btn-group o-add-data-button" ng-right-click="ctrl.openMenu()">
-  <button type="button" class="btn btn-primary col-auto" ng-click="ctrl.notifyApplyPreset()"
-    title="Apply the layer style preset">
-    <i class="fa fa-check"></i>
+  <button type="button" class="btn btn-primary col-auto" 
+      ng-click="ctrl.notifyApplyPreset()"
+      title="Apply the layer style preset">
+    <i class='fa fa-check'></i>
```
Comment (jsalankey) [inline src/os/layer/preset/presetmenubutton.js:546]: Nit: single quotes in an HTML template. I thought the HTML linter disallowed this...

File: src/os/layer/preset/presetmenubutton.js (hunk 2 above)
Comment (glennkitchellcaci) [reply src/os/layer/preset/presetmenubutton.js:301]: :point_up: Magic. I had moved one of these then's into a function at one point during these changes and discovered those ones needed the .bind(this), but these arrow functions are OK.

File: src/os/layer/preset/presetmenubutton.js (hunk 3 above)
Comment (glennkitchellcaci) [reply src/os/layer/preset/presetmenubutton.js]: Updated. Ironically, msg was basically the same thing, so I just used it instead, lol.

**Review by jsalankey — APPROVED** [review]

**Pull request merged**